### ISSUE TITLE
Additional domains for accessing organisation login

### ISSUE THREAD

**Issue by hifabienne** [zitadel/zitadel#5094]
As a developer, I want to be able to run ZITADEL on multiple domains/sub domains and connect an organisation to a specific domain, so if my customers open a domain of an organisation they will see the configured login of there organisation.
E.g: login.myorg.com --> shows instance login, customer-a.myorg.com --> shows login of organisation "customer a"

**Acceptance criteria**
- [ ] I am able to configure multiple domains and subdomains for ZITADEL
- [ ] I am able to connect a specific domain to an organisation (like if I send the organisation scope)
- [ ] When calling a domain with a connected organisation I see the settings configured on the organisation

**Comment by honzapatCZ** [comment]
Wouldnt it make more sense if all registration related stuff were done based on the domain? Matched reserved domain -> registered into system under that organization(if registrations are allowed), not matched -> registered under default organization(if allowed by default organization and/or instance settings)

**Comment by fforootd** [comment]
> Wouldnt it make more sense if all registration related stuff were done based on the domain? Matched reserved domain -> registered into system under that organization(if registrations are allowed), not matched -> registered under default organization(if allowed by default organization and/or instance settings)

Hm that's for sure food for thought. Currently we have two ways of looking onto this:

A) Assigned/Verified Domain - A domain that is being used from an organization and should be allowed to register a user as such.
B) The Domain(s) where ZITADEL is responding to http request

I think @hifabienne wants to combine the two. So that a "verified domain" can also be used as domain to serve the login page.

**Comment by honzapatCZ** [comment]
I'm imagining all this like the Microsoft scenario, which is both B2C and B2B.
If you register by default you are registering into the Microsoft namespace and you are a totally normal customer.
If you however register with an associated domain, you are automatically onboarded into your organization, logos are switched out, etc.

This is what "Organizations" seem to me as opossed to total separation/tennancy.

**Comment by honzapatCZ** [comment]
This way organizations would not change if people can register(the organization selector seems to me like an invitation to attack), but only set if users can auto-onboard from their email domains.

**Comment by fforootd** [comment]
> I'm imagining all this like the Microsoft scenario, which is both B2C and B2B. If you register by default you are registering into the Microsoft namespace and you are a totally normal customer. If you however register with an associated domain, you are automatically onboarded into your organization, logos are switched out, etc.
> 
> This is what "Organizations" seem to me as opossed to total separation/tennancy.

Got it, that makes sense!

**Comment by mffap** [comment]
+ 1 from a customer call

Use case here is as follows: It's a b2b mobile application. The users should be able to login within the context of their organization, meaning they must not use the domain ending behind their name. This option would help to start in that context already by picking the right domain and then applying the org-id scope.

**Comment by mffap** [comment]
+1 from a customer request

**Comment by mffap** [comment]
@hifabienne and @fforootd what is the recommended workaround with ZITADEL Cloud for the moment? 

I guess the customer has to send the org scope with the request to trigger the correct organization. How can they do that when using a domain as entry? I imagine that the IdP should not be the entry point, but rather the application. In that sense the application should already send the org scope and then redirect to the IdP that answers under one of the custom domains.

Any other ideas?

**Comment by hifabienne** [comment]
> @hifabienne and @fforootd what is the recommended workaround with ZITADEL Cloud for the moment?
> 
> I guess the customer has to send the org scope with the request to trigger the correct organization. How can they do that when using a domain as entry? I imagine that the IdP should not be the entry point, but rather the application. In that sense the application should already send the org scope and then redirect to the IdP that answers under one of the custom domains.
> 
> Any other ideas?

Yes exactly, I assume if they want to have different domains for the login they will also have different domains for the app. In that case they could send the org scope as soon as they redirect. they just need to have the org id stored somewhere on their side.

**Comment by schettn** [comment]
This is very useful in terms of password managers.

Currently my password manager (Keychain) cannot differentiate between different organisations.

So when I save my credentials (same email address but different passwords) for org1 and later for org2 then org1 gets overwritten.

This is because password managers do not take the complete URL but only the origin.

This is not a problem when I use the username including the org domain suffix but that is not feasible.

**Comment by elinashoko** [comment]
Thanks for filing this issue. As part of an ongoing backlog cleanup, we're closing issues that have had no activity in over 12 months.

If this is still relevant, let us know in a comment (or reopen the issue) and we'll take another look.

This issue will close in 30 days if there's no further activity.
